### PR TITLE
Add support for Rails 8, use Appraisal to test gem against older Rails versions

### PR DESCRIPTION
**Pull request by n-rodriguez**
Thank you!

### CODE REVIEW

**Comment by n-rodriguez** [comment]
ping @strzibny

**Comment by n-rodriguez** [comment]
This PR fixes the error in CI https://github.com/wmlele/devise-otp/actions/runs/11869310036

File: test/dummy/db/migrate/20240604000001_create_admins.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-class CreateAdmins < ActiveRecord::Migration[7.1]
+class CreateAdmins < ActiveRecord::Migration[5.0]
```
Comment (strzibny) [inline test/dummy/db/migrate/20240604000001_create_admins.rb:1]: Why this change?

**Comment by strzibny** [comment]
Great, thx!
